third-party-code.rst

```diff
@@ -0,0 +1,253 @@
+   Refers to the code behind the PHP.net website, the documentation generator
+   project PhD, the PHP wiki, and other similar systems. In general, “PHP code
+   run by PHP.net.”
+
+**Documentation**
+   Refers to objective information about PHP, the PHP language, the PHP standard
+   library, and PHP ecosystem hosted on PHP.net. This may include reference
+   material, tutorials, FAQs, and similar.
+
+**Marketing material**
+   Refers to content on PHP.net or similar sites intended to promote or
+   evangelize PHP the language or ecosystem.
+
+**Libraries**
+   Refers to existing third party code packages or tools, either C extensions or
+   PHP code, maintained by someone other than the PHP Internals team. It also
+   includes command line utilities used primarily by a developer. It may also
+   refer to non-profit PHP ecosystem organizations, such as the PHP Foundation
+   or PHP-FIG.
+
+**Web Application**
+   Refers to a “full” web framework that provides end-to-end web application
+   capabilities, or an installable complete application. It does not refer to
+   command line utilities used primarily by developers building applications.
+
+**Approved license**
+   Refers to a license `approved by the Free Software Foundation as Free
+   Software <https://www.gnu.org/licenses/license-list.en.html>`_, and that is
+   inbound compatible with GPLv3.
+
+*********************
+ Approval Heuristics
+*********************
+
+PHP Tooling
+===========
+
+PHP tooling MAY make use of third party libraries, provided that the library
+meets all of the “Inclusion” criteria, and does not meet any of the “Exclusion”
+criteria.
+
+Inclusion criteria
+------------------
+
+#. The library must have a stable >= 1.0 release, and have had one for at least
+   a year. (This is to ensure it has longevity.)
+#. The library provides targeted, necessary functionality.
+#. The library is a recognized de facto standard, or one of a small number of de
+   facto standards, in its problem space.
+#. The library is available under an Approved License.
+
+Exclusion criteria
+------------------
+
+#. The library is a Web Application
+#. The library is not available under an Approved License.
+#. The library has shown no meaningful activity for one year prior to its first
+   inclusion.
+
+PHP tooling maintainers MAY use their judgement to determine if a library meets
+the above criteria, but SHOULD be conservative in their interpretation of
+whether or not a library satisfies the necessary criteria.
+
+Explicitly approved libraries
+-----------------------------
+
+The following libraries have been explicitly approved by RFC vote.
+
+-  Composer
+-  Xdebug
+-  phpunit/phpunit
+-  phpstan/phpstan
+-  vimeo/psalm
+-  michelf/php-markdown
+-  phpmailer/phpmailer
+-  squizlabs/php_codesniffer
+-  friendsofphp/php-cs-fixer
+-  symfony/dotenv
+-  symfony/console
+-  fzaninotto/faker
+-  erusev/parsedown
+-  amenadiel/jpgraph
+-  Any library or PSR published by the PHP-FIG
```

Comment:
That's a pretty big door ... PHP-FIG could publish a PSR abolishing the RFC process, and it would have to be adopted per this text.

Reply:
No, because Library is defined as code, or the existence of an org.  It's also optional.  This is saying that if, for instance, we wanted to use fig/log-util on PHP.net, or mention the existence of PER-CS on the site, we can.  Nothing in this policy document says we MUST do so.

Reply:
As I mentioned on the internals list, I still fail to see why we need this exception. The rules for inclusion and exclusion are clear and objective. As far as I can tell, `fig/log-util` fits the criteria, so it could already be used without this line. Can you clarify why this is needed? What happens if a library published by fig in the future doesn't meet the inclusion criteria?

Reply:
> No, because Library is defined as code, or the existence of an org.

Then it should just say "library" and not PSR. PSR doesn't look like it explicitly means "code" or "library." There are PSRs that have nothing to do with code or libraries at all. Does this mean PHP is required to adapt them if this is merged? If not, this should be more explicitly defined as how it is worded now, doesn't restrict it.

Reply:
I'd also add that a number of PSRs probably shouldn't be included out-right. There are some that are actually quite bad for PHP, due to the existence of modern PHP, and each should be considered separately.

Reply:
Again, you keep using the word "required."  Nothing is required.  This policy proposal puts zero requirements on the PHP project.  It *allows* people working on the support structure around it to use and talk about things, but there is no requirement that anyone mention anything.  Literally the only place the word "required" appears on this page is in your comment (and now this comment).

That you keep implying something that is not there, and which I already clarified, strikes me as rather disingenuous.

Reply:
Ah. I misread the paragraph preceding it. Sorry for the drive-by comment then.